src/github.com/getlantern/flashlight/config/fetcher.go

```diff
@@ -117,8 +119,8 @@ func (cf *fetcher) fetchCloudConfig(cfg *Config) ([]byte, error) {
 	req.Header.Set("Lantern-Fronted-URL", cfg.FrontedCloudConfig+cb)
 
 	id := cf.user.GetUserID()
-	if id != 0 {
-		req.Header.Set(userIDHeader, string(id))
+	if id != "" {
+		req.Header.Set(userIDHeader, id)
```

Comment:
Heart of the fix. Just send in the string if it's there without worrying about conversions.